app/Platform/Actions/UpdateAchievementMetricsAction.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Models\Achievement;
+use App\Models\Game;
+use App\Models\PlayerAchievement;
+use App\Platform\Services\SearchIndexingService;
+use Illuminate\Support\Collection;
+
+class UpdateAchievementMetricsAction
+{
+    public function execute(Achievement $achievement): void
+    {
+        $this->update($achievement->game, collect([$achievement]));
+    }
+
+    /**
+     * @param Collection<int, Achievement> $achievements
+     */
+    public function update(Game $game, Collection $achievements): void
```

Comment:
alternate entry point allows updating all of the published achievements for the game at once.